File: z-clients/javascript/src/apis/msgsSvixPoller.ts
// this file is @generated

import {
    type ListResponseSvixPollerOut,
    ListResponseSvixPollerOutSerializer,
} from '../models/listResponseSvixPollerOut';
import {
    type SvixPollerCreateIn,
    SvixPollerCreateInSerializer,
} from '../models/svixPollerCreateIn';
import {
    type SvixPollerCreateOut,
    SvixPollerCreateOutSerializer,
} from '../models/svixPollerCreateOut';
import {
    type SvixPollerDeleteIn,
    SvixPollerDeleteInSerializer,
} from '../models/svixPollerDeleteIn';
import {
    type SvixPollerDeleteOut,
    SvixPollerDeleteOutSerializer,
} from '../models/svixPollerDeleteOut';
import {
    type SvixPollerListIn,
    SvixPollerListInSerializer,
} from '../models/svixPollerListIn';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class MsgsSvixPoller {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Create a Svix poller configuration for a topic. */
    public create(
        svixPollerCreateIn: SvixPollerCreateIn,
    ): Promise<SvixPollerCreateOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.msgs.svix-poller.create");

        request.setBody(
            SvixPollerCreateInSerializer._toJsonObject(svixPollerCreateIn)
        );
        
        return request.send(
            this.requestCtx,
            SvixPollerCreateOutSerializer._fromJsonObject,
        );
    }/** Delete a Svix poller configuration. */
    public delete(
        svixPollerDeleteIn: SvixPollerDeleteIn,
    ): Promise<SvixPollerDeleteOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.msgs.svix-poller.delete");

        request.setBody(
            SvixPollerDeleteInSerializer._toJsonObject(svixPollerDeleteIn)
        );
        
        return request.send(
            this.requestCtx,
            SvixPollerDeleteOutSerializer._fromJsonObject,
        );
    }/** List Svix poller configurations for a topic. */
    public list(
        topic: string,
        svixPollerListIn: SvixPollerListIn,
    ): Promise<ListResponseSvixPollerOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.msgs.svix-poller.list");

        request.setBody(
            SvixPollerListInSerializer._toJsonObject({
                ...svixPollerListIn,
                topic: topic,
            })
        );
        
        return request.send(
            this.requestCtx,
            ListResponseSvixPollerOutSerializer._fromJsonObject,
        );
    }
}

File: z-clients/java/src/main/java/com/svix/diom/apis/MsgsSvixPoller.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.DiomException;
import com.svix.diom.HttpClient;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.ListResponseSvixPollerOut;
import com.svix.diom.models.SvixPollerCreateIn;
import com.svix.diom.models.SvixPollerCreateOut;
import com.svix.diom.models.SvixPollerDeleteIn;
import com.svix.diom.models.SvixPollerDeleteOut;
import com.svix.diom.models.SvixPollerListIn;
import com.svix.diom.models.SvixPollerListIn_;

public class MsgsSvixPoller {
    private final HttpClient client;

    public MsgsSvixPoller(HttpClient client) {
        this.client = client;
    }

    /** Create a Svix poller configuration for a topic. */
    public SvixPollerCreateOut create(
        final SvixPollerCreateIn svixPollerCreateIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.msgs.svix-poller.create",
            null,
            svixPollerCreateIn,
            SvixPollerCreateOut.class
        );
    }

    /** Delete a Svix poller configuration. */
    public SvixPollerDeleteOut delete(
        final SvixPollerDeleteIn svixPollerDeleteIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.msgs.svix-poller.delete",
            null,
            svixPollerDeleteIn,
            SvixPollerDeleteOut.class
        );
    }

    /** List Svix poller configurations for a topic. */
    public ListResponseSvixPollerOut list(
        String topic,
        final SvixPollerListIn svixPollerListIn
    ) throws DiomException {
        SvixPollerListIn_ body = new SvixPollerListIn_(
            svixPollerListIn.getNamespace(),
            topic,
            svixPollerListIn.getLimit(),
            svixPollerListIn.getIterator()
        );

        return this.client.executeRequest(
            "POST",
            "/api/v1.msgs.svix-poller.list",
            null,
            body,
            ListResponseSvixPollerOut.class
        );
    }

    /** List Svix poller configurations for a topic. */
    public ListResponseSvixPollerOut list(
        String topic
    ) throws DiomException {
        return this.list(
            topic,
            new SvixPollerListIn()
        );
    }
}